protocol/0078-NWLI-network_wide_limits.md

```diff
@@ -0,0 +1,80 @@
+
+## Number of assets
+
+`limits.assets.max`
+
+TODO: Do we really care? Each asset is added via a governance vote, so it doesn't seem likely we'll get a ton of unnecessary ones (plus we'd then have to implement a process of removing an asset with a vote if the limit was already reached, but there was some new asset that community perceived as very important). Also, does an additional asset itself really increase the load that much?
+
+## Number of markets
+
+Introduce a new network parameter `limits.markets.max` controlling the maximum number of markets allowed within a network.
+
+Each market in a `Proposed` [state](./0043-MKTL-market_lifecycle.md) should increment the **total market count** by 1. If a market reaches a `Rejected`, `Cancelled`, `Closed` or `Settled` state the total market count should be decremented by 1.
+As soon as the total market count reaches the value specified by `limits.markets.max` parmeter no other further market proposals should be accepted. If the total market count drops below `limits.markets.max` the network should start accepting market proposals again.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total market count then no further actions are needed. Specifically: all existing markets continue functioning in their respective states, but no further proposals are accepted until the count drops below the new limit.
+
+## Parties in a market
+
+Introduce a new network parameter `limits.markets.maxParties` controlling the maximum number of parties allowed within any given market.
+
+A party gets counted towards the limit if it has either open orders or open positions in the market. Once party has no open orders and no open positions it gets removed from a **total party count** within the market. Once the limit gets reached the market accepts no further orders.
+Once the total party count drops below the limit market accepts orders again (provided its [state](./0043-MKTL-market_lifecycle.md) allows that).
+
+The limit does not apply to liquidity providers.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total party count in any given market then no further actions are needed. Specifically: all parties in that market remain unaffected (their positions and orders remain open), but no new parties are allowed to participate in the market until the total party count drops below the new limit.
+
+## Limit orders on a book
+
+Introduce a new network parameter `limits.markets.maxLimitOrders` controlling the maximum number of limit orders that can rest on a book in any given market. If the limit is reached limit orders can still be submitted, but:
```

Comment:
A limit order of arbitrary volume contributes +1 to the count.

Reply:
Added now.